truffle/com.oracle.truffle.api.instrumentation/src/com/oracle/truffle/api/instrumentation/TruffleInstrument.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (c) 2016, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.truffle.api.instrumentation;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import com.oracle.truffle.api.CallTarget;
+import com.oracle.truffle.api.source.Source;
+
+/**
+ * <p>
+ * This instrumentation service provider interface (SPI) provides a way to observe and inject
+ * behavior into interpreters written using the Truffle framework. A registered instrumentation can
+ * get created and disposed mulitple times by the runtime system. But only one instance is active
+ * per runtime system. When the instrumentation gets created then
+ * {@link #onCreate(Env, Instrumenter)} and when it is disposed {@link #onDispose(Env)} gets
+ * invoked. Instrumentations can bind listeners to the execution of guest languages by using the
+ * {@link Instrumenter instrumenter} class that is passed as parameter
+ * {@link #onCreate(Env, Instrumenter)}. For each instrumentation instance
+ * {@link #onCreate(Env, Instrumenter)} and {@link #onDispose(Env)} is invoked exactly once.
```

Comment:
First paragraph is awkward and a bit redundant with respect to the lifecycle of an instance; can you make it more clear?